Test for CPL_DEBUG=ON failing in GDAL 1.x

The most recent commit is causing a test to fail in GDAL 1.x.

https://travis-ci.org/Toblerity/Fiona/builds/359549791

https://github.com/Toblerity/Fiona/compare/35bdccb3942f...8282a8d8da6f

```
_________________________________ test_options _________________________________
tmpdir = local('/tmp/pytest-of-travis/pytest-0/test_options0')
    def test_options(tmpdir):
        """Test that setting CPL_DEBUG=ON works"""
        logfile = str(tmpdir.mkdir('tests').join('test_options.log'))
        logger = logging.getLogger()
        logger.setLevel(logging.DEBUG)
        fh = logging.FileHandler(logfile)
        fh.setLevel(logging.DEBUG)
        logger.addHandler(fh)
    
        with fiona.drivers(CPL_DEBUG=True):
            path = os.path.join("tests", "data", "coutwildrnp.shp")
            c = fiona.open(path)
            c.close()
            with open(logfile, "r") as f:
                log = f.read()
>           assert "GDALClose" in log
E           AssertionError: assert 'GDALClose' in 'Creating a not-responsible GDALEnv in drivers()\nGDAL_DATA: /home/travis/gdalinstall/gdal-1.11.5/share/gdal\nPROJ_LIB... in OGR: OGROpen(tests/data/coutwildrnp.shp/0x20d2790) succeeded as ESRI Shapefile.\nFlushed buffer\nStopped session\n'
tests/test_drivers.py:30: AssertionError
```

@sgillies

--- a/tests/test_drivers.py
+++ b/tests/test_drivers.py
@@ -11,6 +11,7 @@ import fiona
 
 logging.basicConfig(stream=sys.stderr, level=logging.DEBUG)
 
+GDAL_MAJOR_VER = fiona.get_gdal_version_num() // 1000000
 
 def test_options(tmpdir):
     """Test that setting CPL_DEBUG=ON works"""
@@ -27,4 +28,7 @@ def test_options(tmpdir):
         c.close()
         with open(logfile, "r") as f:
             log = f.read()
-        assert "GDALClose" in log
+        if GDAL_MAJOR_VER >= 2:
+            assert "GDALOpen" in log
+        else:
+            assert "OGROpen" in log

